configs/sites/tier2/macos.default/packages.yaml

```diff
@@ -1,6 +1,9 @@
 packages:
   cairo:
     variants: ~png ~svg
+  # https://github.com/JCSDA/spack-stack/issues/1211
+  fms:
+    variants: +shared
```

Comment:
It appears that this variants spec is overriding the fms.variants spec in common/packages.yaml. This is leading to the mac building only the 32 bit library (since 32 is the default setting for the precision variant). Here is the concretized fms spec before this change:
```
-   gajprcw      ^fms@2024.02%apple-clang@15.0.0 ldflags=-Wl,-ld_classic +deprecated_io+gfs_phys+internal_file_nml~ipo~large_file+openmp+pic+quad_precision+sharedlibs~yaml build_system=cmake build_type=Release constants=GFS generator=make patches=b9a9bb6 precision=32,64 arch=darwin-sonoma-m1
```
and here is the spec after this change:
```
 -   7ixjuiy      ^fms@2024.02%apple-clang@15.0.0 ldflags=-Wl,-ld_classic ~deprecated_io+gfs_phys+internal_file_nml~ipo~large_file+openmp~pic+quad_precision+shared~yaml build_system=cmake build_type=Release constants=GFDL generator=make patches=2b12a6c precision=32 arch=darwin-sonoma-m1
```
Note that the constants spec changed from GFS to GFDL too.

Is there a syntax that tells spack to merge this spec with the variants spec in common/packages.yaml (instead of overriding)?

Reply:
Good catch! Yes, there is. Let me try to fix this.